pkg/sm/sm.go

```diff
@@ -460,13 +460,20 @@ func (smb *ServiceManagerBuilder) EnableMultitenancy(labelKey string, extractTen
 
 	multitenancyFilters := filters.NewMultitenancyFilters(labelKey, extractTenantFunc)
 	smb.RegisterFiltersAfter(filters.ProtectedLabelsFilterName, multitenancyFilters...)
-	smb.RegisterPlugins(osb.NewCheckInstanceOwnerPlugin(smb.Storage, labelKey))
+	smb.RegisterFilters(
```

Comment:
These are applicable not only when `EnableMultitenancy` is called. This is a central feature that must work even if and multi-tenancy is not enabled. I don't think multi-tenancy is the correct term any more. This should be promoted to a functionality that is enabled by default and it should require a configuration.